pkg/cmd/target/target.go

```diff
@@ -128,23 +141,21 @@ func ValidateKind(kind TargetKind) error {
 	return fmt.Errorf("invalid target kind given, must be one of %v", AllTargetKinds)
 }
 
-func validTargetArgsFunction(f util.Factory, o *TargetOptions, args []string, toComplete string) ([]string, error) {
-	if len(args) == 0 {
-		return []string{
-			string(TargetKindGarden),
-			string(TargetKindProject),
-			string(TargetKindSeed),
-			string(TargetKindShoot),
-			string(TargetKindPattern),
-			string(TargetKindControlPlane),
-		}, nil
-	}
+type cobraValidArgsFunction = func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective)
 
-	kind := TargetKind(strings.TrimSpace(args[0]))
-	if err := ValidateKind(kind); err != nil {
-		return nil, err
+func validTargetFunctionWrapper(f util.Factory, ioStreams util.IOStreams, kind TargetKind) cobraValidArgsFunction {
```

Comment:
For this PR I would keep it like it is now and not introduce the anonymous func.